op-deployer/pkg/deployer/state/overrides.go

```diff
@@ -0,0 +1,98 @@
+// File: op-deployer/pkg/deployer/state/valid_overrides.go
+
+package state
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// ValidOverrideKeys is a map of all keys that can be overridden in the deployment config
+var ValidOverrideKeys = map[string]struct{}{
```

Comment:
Do we need to make this more specific? i.e. dependent on the intent type (standard, vs standard-with-overrides, vs custom) different fields can be overriden